src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -259,34 +256,37 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, ServerOpt
         /// <returns>The <see cref="ListenOptions"/>.</returns>
         public static ListenOptions UseHttps(this ListenOptions listenOptions, ServerOptionsSelectionCallback serverOptionsSelectionCallback, object state, TimeSpan handshakeTimeout)
         {
-            // HttpsOptionsCallback is an internal delegate that is the ServerOptionsSelectionCallback, a ConnectionContext, and the ClientCertificateMode.
-            // Given that ConnectionContext will eventually be replaced by System.Net.Connections, it doesn't make much sense to make the HttpsOptionsCallback delegate public.
-            return listenOptions.UseHttps(GetTlsOptionsAsync, state, handshakeTimeout);
-
-            async ValueTask<(SslServerAuthenticationOptions, ClientCertificateMode)> GetTlsOptionsAsync(
-                ConnectionContext connection, SslStream stream, SslClientHelloInfo clientHelloInfo, object state, CancellationToken cancellationToken)
+            return listenOptions.UseHttps(new TlsHandshakeCallbackOptions(context =>
             {
-                var tlsOptions = await serverOptionsSelectionCallback(stream, clientHelloInfo, state, cancellationToken);
-                return new (tlsOptions, ClientCertificateMode.DelayCertificate);
-            }
+                context.AllowDelayedClientCertificateNegotation = true; // TODO: do we still need this enabled by default after adding the new overload?
```

Comment:
Don't we want to discourage the usage of delayed cert negotiation? So we should have it off by default IMO.

Reply:
If we turn it off here then it won't be usable at all from this UseHttps overload. That's probably OK.